New Feature: Create Context Menu on text selection

nirajkaushal opened the pull request:
This commit add context menu when user select some text and make right-click.
User can search by clicking on that menu.

mrshu commented on js/background.js:
```diff
@@ -85,3 +85,17 @@ chrome.omnibox.onInputEntered.addListener( function(text) {
+//This adds Context Menu when user select some text.
+//create context menu
+chrome.contextMenus.create({
+    title : 'Search anonymously on DuckDuckGo',
```
@nirajkaushal would it be possible to use the `%s` placeholder here as described in [the docs](https://developer.chrome.com/extensions/contextMenus#property-create-createProperties)? That way we can use the same `Search DuckDuckGo for "%s"` here as we do in Firefox for instance

![](https://duckduckgo.com/extensions/thanks/images/context-search.png)

nirajkaushal replied:
Yes, it is possible to add selection text in context menu. I am requested for new PR

mrshu commented on js/background.js:
```diff
@@ -85,3 +85,17 @@ chrome.omnibox.onInputEntered.addListener( function(text) {
+});
+
+//execute when user click on that context menu
+function searchThis(selectedText){
```
@nirajkaushal is there any reason why this separate function needs to be created? Could it be made an anonymous function on line 94?

nirajkaushal replied:
I added anonymous function instead of separate function in new PR

mrshu commented:
Thanks for this PR @nirajkaushal, it is awesome to see that we might have the same feature set on all platforms (Firefox and Safari already have context menu text selection capability of sorts).

nirajkaushal commented:
Thanks for merge @mrshu , I have updated title of context menu as you expected and also replaced separate function with anonymous function of onclick event.

mrshu commented:
@nirajkaushal Thanks a ton, this is ready to be merged!

Would you mind [squashing these commits](http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html) into one so that we do not pollute the git log?

Once you do that I will merge your PR at once.

nirajkaushal commented:
I squashed these commits into one commit 3e791f9.

mrshu commented:
@nirajkaushal Beautiful, thanks a ton for your contribution again.

This LGTM.

nirajkaushal commented:
Thanks for merge @mrshu
